pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -147,3 +171,21 @@ func mergeTags(tags ...map[string]string) (result map[string]*string) {
 		return strings.ReplaceAll(key, "/", "_"), to.StringPtr(value)
 	})
 }
+
+// getVnetInfoLabels returns the labels for Azure CNI
+func (p *Provider) getVnetInfoLabels(ctx context.Context, _ *v1alpha2.AKSNodeClass) (map[string]string, error) {
+	// TODO(bsoghigian): this should be refactored to lo.Ternary(nodeClass.Spec.VnetSubnetID != nil, lo.FromPtr(nodeClass.Spec.VnetSubnetID), os.Getenv("AZURE_SUBNET_ID")) when we add VnetSubnetID to the nodeclass
+	vnetSubnetComponents, err := utils.GetVnetSubnetIDComponents(options.FromContext(ctx).SubnetID)
+	if err != nil {
+		return nil, err
+	}
+	vnetLabels := map[string]string{
+		vnetDataPlaneLabel:      networkDataplaneCilium,
+		vnetNetworkNameLabel:    vnetSubnetComponents.VNetName,
+		vnetSubnetNameLabel:     vnetSubnetComponents.SubnetName,
+		vnetSubscriptionIDLabel: vnetSubnetComponents.SubscriptionID,
+		vnetGUIDLabel:           p.vnetGUID,
+		vnetPodNetworkTypeLabel: networkModeOverlay,
```

Comment:
Overlay only needs VNet GUID, subnet name and pod network type. The full set will be needed for pod subnet.